.github/workflows/build-flutter.yml

```diff
@@ -1,121 +1,175 @@
           upload-key-password: ${{ secrets.upload-key-password }}
 
-      - name: Build
-        uses: lenra-io/github-actions/jobs/build-flutter@main
+      - name: Update version
+        uses: actions/github-script@v7
         with:
-          version: ${{ inputs.version }}
-          build: ${{ inputs.build }}
-          platform: ${{ inputs.platform }}
-          targets: ${{ inputs.targets }}
-          flutter-extra-args: ${{ inputs.flutter-extra-args }}
+          script: |
+            // Update pubspec.yaml inplace
+            const fs = require('fs')
+            const path = require('path')
+            const pubspecPath = path.join(process.env.GITHUB_WORKSPACE, 'pubspec.yaml')
+            // edit version and build number
+            const pubspec = fs.readFileSync(pubspecPath, 'utf8')
+    
+            # // Remove channel from version, if we have `1.0.0-beta.1`, we want `1.0.0.1`
+            # const version = '${{ inputs.version }}'.replace(/-.*$/, '')
+            const version = '${{ inputs.version }}'
+    
+            const build = '${{ inputs.build }}'
+            let version_string = ''
+            if (build.length > 0) {
+              version_string = `${version}+${build}`
+            } else {
+              version_string = `${version}`
+            }
+            const newPubspec = pubspec.replace(/version: .*/, `version: ${version_string}`)
+            fs.writeFileSync(pubspecPath, newPubspec)
+
+      - name: Install Dependencies (linux)
+        shell: sh
+        if: contains(inputs.targets, 'appimage')
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y make cmake ninja-build libgtk-3-dev locate libfuse2 libsecret-1-dev libjsoncpp-dev
+          wget -O appimagetool "https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage"
+          chmod +x appimagetool
+          mv appimagetool /usr/local/bin/
+    
+      - name: Install Dependencies (macos)
+        shell: sh
+        if: contains(inputs.targets, 'dmg')
+        run: |
+          npm install -g appdmg
+
+      - name: Build App for ${{ inputs.platform }}
+        run: |
+          echo "Building for platform: ${{ inputs.platform }}"
+          echo "Version: ${{ inputs.version }}"
+          echo "Build ID: ${{ inputs.build }}"
+          echo "Targets: ${{ inputs.targets }}"
+
+          dart pub global activate flutter_distributor
+
+          platform=${{ inputs.platform }}
+          IFS=',' read -r -a flutter_extra_args <<< "${{ inputs.flutter-extra-args }}"
+          IFS=',' read -r -a targets <<< "${{ inputs.targets }}"
+
+          if [ "${#targets[@]}" -gt 0 ]; then
+            echo "No targets specified. Exiting..."
+            exit 1
+          fi
+
+          # If flutter_extra_args has split-per-abi,
+          # separate APK build than other targets 
+          # and use split-per-abi only on android apk build.
+          if [ "${platform}" = "android" ]; then
```

Comment:
I don't understand why this is done ?

You should remove it, we won't use it for our project